TombEngine/Scripting/Internal/TEN/Misc/Miscellanous.cpp

```diff
@@ -332,6 +333,31 @@ namespace Misc
 		ObjCamera(LaraItem, 0, LaraItem, 0, false);
 	}
 
+	/// Writing messages within the Log file
+	//@function PrintLog
+	//@tparam string message to be displayed within the Log
+	//@tparam Misc.LogLevel logLevel log level to be displayed
+	//@tparam[opt] bool allowSpam true allows spamming of the message
+	// Must be one of:
+	// INFO
+	// WARNING
+	// ERROR
+	// 
+	//@usage
+	//PrintLog('test info log', LogLevel.INFO)
+	//PrintLog('test warning log', LogLevel.WARNING)
+	//PrintLog('test error log', LogLevel.ERROR)
+	// -- spammed message
+	// PrintLog('test spam log', LogLevel.INFO, true)
+	// 
+	// For native Lua handling of errors, see the official guide
+	//<a href="https://www.lua.org/pil/8.3.html">Error management</a>
+	//<a href="https://www.lua.org/manual/5.4/manual.html#pdf-debug.traceback">debug.traceback</a>
+	static void PrintLog(std::string const& message,LogLevel const& level, bool allowSpam = false)
```

Comment:
Use `TypeOrNil<bool> allowSpam` and omit the default arg, and then use `USE_IF_HAVE(bool, allowSpam, false)`. This will tell Sol that the last arg is optional. See EffectsFunctions.cpp for other examples where optional args are set up like this.